instructor/common.py

```diff
@@ -0,0 +1,604 @@
+from django.shortcuts import render_to_response, redirect
+from django.template import RequestContext
+from django import forms
+from django.forms.models import modelformset_factory, inlineformset_factory
+from django.forms.models import modelform_factory
+from instructor import models
+from instructor import compiler
+from django.contrib.auth.models import User
+import datetime
+
+from backend.models import Assignment, StudentAssignment
+
+
+def courses(request):
+    user = request.user
+    message = ''
+    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'], can_delete=True)
+    if request.method == "POST":
+        formset = CourseFormSet(request.POST)
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.owner = request.user
+                form.save()
+        else:
+            message = "Something went wrong"
```

Comment:
you should allow the form errors to be displayed in the templates so that the user has a more actionable error, than just "something went wrong". You should also probably use django messages, but that isn't required